source/_code.html.erb

```diff
@@ -0,0 +1,16 @@
+<div class="container">
+  <button aria-controls="<%= name %>-snippets"
```

Comment:
I see what you're getting at, played around with it some, and it's an interesting thought exercise.

I *think* (at least in this case) that a preferred experience for those using assistive technologies would emulate the experience of those who don't. That way there wouldn't be major confusing discrepancies with screen readers and what's visible on the page.